config/main.py

```diff
@@ -351,88 +351,95 @@ def get_interface_naming_mode():
         mode = "default"
     return mode
 
-def _is_neighbor_ipaddress(ipaddress):
+# Get the local BGP ASN from DEVICE_METADATA
+def get_local_bgp_asn(config_db):
+    metadata = config_db.get_table('DEVICE_METADATA')
+    return metadata['localhost']['bgp_asn']
+
+def _is_neighbor_ipaddress(config_db, ipaddress):
     """Returns True if a neighbor has the IP address <ipaddress>, False if not
     """
-    config_db = ConfigDBConnector()
-    config_db.connect()
     entry = config_db.get_entry('BGP_NEIGHBOR', ipaddress)
     return True if entry else False
 
-def _get_all_neighbor_ipaddresses():
+def _get_all_neighbor_ipaddresses(config_db, ignore_local_hosts=False):
     """Returns list of strings containing IP addresses of all BGP neighbors
+       if the flag ignore_local_hosts is set to True, additional check to see if 
+       if the BGP neighbor AS number is same as local BGP AS number, if so ignore that neigbor.
```

Comment:
so the name of the function is missleading, because you're filtering out peer with local bgp AS?
Also, we could have dynamic bgp neighbors and bgp_monitor entries.
Should we include them here?

Reply:
This was an existing function, to which I added a new parameter ignore_local_hosts to do bgp AS based filtering. I will change the name of the function.
 
Currently all the functions look in the BGP_NEIGHBOR table alone, we need change in all API's to check the other BGP tables as well. BGP DYNAMIC PEERS and BGP MONITORS. We could do it separately in a different PR if that is ok